MigrationScript/Migration.ps1

```diff
@@ -147,4 +157,95 @@ function Main
     Stop-Transcript
 }
 
+<#
+.SYNOPSIS
+Function to migrate the Hana Providers to ams v2
+
+.PARAMETER secretName
+Secret Name, this contains the name of the provider. 
+
+.PARAMETER secretValue
+Secret Value, this contains the provider configuration.
+
+.PARAMETER logger
+logger object.
+
+.EXAMPLE
+MigrateHanaProvider -secretName $secret.name -secretValue $secret -logger $logger
+#>
+function MigrateHanaProvider([string]$secretName, $secretValue, $logger) {
+	$parsedArmId = Get-ParsedArmId $amsv2ArmId
+
+	[string]$monitorName = $parsedArmId.amsResourceName;
+	[string]$resourceGroupName = $parsedArmId.amsResourceGroup;
+	[string]$subscriptionId = $parsedArmId.subscriptionId;
+	# set context.
+	Set-AzContext -SubscriptionId $subscriptionId -TenantId $tenantId;
+
+	[string]$providerType = $($secretValue.type);
+	[string]$providerName = $($secretValue.name) # + (Get-Random -Minimum 2 -Maximum 20000).ToString();
+	Write-Host "Provider Name is $($secretValue.name)";
+	$providerProperties = $($secretValue.properties)
+	$requestObj = @{
+		Name = $providerName
+		body = @{
+			properties = @{
+				providerSettings = @{
+					providerType = $providerType
+					hostname = $($providerProperties.hanaHostname)
+					dbName = $($providerProperties.hanaDbName)
+					sqlPort = $($providerProperties.hanaDbSqlPort).ToString()
+					dbUsername = $($providerProperties.hanaDbUsername)
+					dbPassword = $($providerProperties.hanaDbPassword)
+				}
+			}
+		}
+	}
+
+	# check provider status before making a put call
+	$getResponse = GetAmsV2ProviderStatus -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -providerName $providerName -logger $logger;
```

Comment:
Do we need to check the status code ? 404 etc ?

Reply:
only in case of provisioning state successful, we need to check, for rest all the cases currently we are trying to make the put call again, for 404 Resource not found, we are proceeding with the put call

Reply:
Makes sense.